enhancements/cluster-capabilities-api.md

```diff
@@ -0,0 +1,357 @@
+operators will re-configure their operands.
+
+For a downgrade, the administrator may need to manually update the
+`capabilities` API to change settings that use new values not
+supported in the older version.
+
+### Version Skew Strategy
+
+Clusters deployed using 4.7 or earlier will not have the capabilities
+CRD or resource in place. We have identified two basic approaches to
+address this:
+
+1. We could provide a library for operators to use to retrieve the
+   "current" capabilities settings. That library could provide error
+   handling, so that when the CRD or CR are not found then default
+   values can be returned in a consistent way.
+2. We could introduce the new CRD and a default CR into a 4.7 z-stream
+   release. The CRD and CR would not be used by any 4.7 operators, but
+   by introducing them early we could ensure that they are present
+   before an upgrade to 4.8 where the operators would expect to be
+   able to find them.
+
+Option 1 appears to be less work initially, but might mean that
+long-lived clusters never have a `capabilities` CR created, which
+would mean we have to maintain the library code indefinitely. Option 2
+would mean more coordinated work initially, and would introduce a
+choke point into the supported upgrade graph. A hybrid approach of
+using option 1 for 4.8 and having an operator
+(`cluster-config-operator`?) in 4.8 create the CR if it is not present
+might be best.
+
+*Input needed to make the right choice.*
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+The idea is to find the best form of an argument why this enhancement should _not_ be implemented.
+
+## Alternatives
+
+### Add fields to existing configuration APIs
+
+We could add the fields we know about today to the
+`infrastructure.config.openshift.io` API, instead of creating a new
+API. This would be less work, initially, but muddies the meaning of
+the APIs.
+
+### Expose the cluster profile name directly
+
+We could add a `clusterProfile` field to an existing or new
+configuration API. This would require each operator to interpret each
+new profile, which would make maintaining profiles even more work than
+it is now. It may also encourage us to create more profiles, when we
+have stated elsewhere that our goal is to have as few as possible.
+
+### Use the cluster profile to have different Deployments for operators
+
+We could use different manifests to have the cluster-version-operator
+deploy operators with different configuration (via command line
+arguments or environment variables). This would give us a way to
+configure operators differently for each profile, but those settings
+would be less discoverable and we would not be able to offer users a
+way to change them at runtime (if we ever want to do that).
+
+### Recognize HA/non-HA configuration based on control plane node count
```

Comment:
this is also confusing with externalized control plane, where you have no control plane hosts in the cluster, and a variable number of workers to meet the workload demand.  prefer projecting via the Infrastructure CR.
//cc @csrwng

Reply:
Good point, I'll add this to the doc.